rest_framework_json_api/serializers.py

```diff
@@ -163,3 +167,140 @@ def get_field_names(self, declared_fields, info):
                 declared[field_name] = field
         fields = super(ModelSerializer, self).get_field_names(declared, info)
         return list(fields) + list(getattr(self.Meta, 'meta_fields', list()))
+
+
+class PolymorphicSerializerMetaclass(SerializerMetaclass):
+    """
+    This metaclass ensures that the `polymorphic_serializers` is correctly defined on a
+    `PolymorphicSerializer` class and make a cache of model/serializer/type mappings.
+    """
+
+    def __new__(cls, name, bases, attrs):
+        new_class = super(PolymorphicSerializerMetaclass, cls).__new__(cls, name, bases, attrs)
+
+        # Ensure initialization is only performed for subclasses of PolymorphicModelSerializer
+        # (excluding PolymorphicModelSerializer class itself).
+        parents = [b for b in bases if isinstance(b, PolymorphicSerializerMetaclass)]
+        if not parents:
+            return new_class
+
+        polymorphic_serializers = getattr(new_class, 'polymorphic_serializers', None)
+        if not polymorphic_serializers:
+            raise NotImplementedError(
+                "A PolymorphicModelSerializer must define a `polymorphic_serializers` attribute.")
+        serializer_to_model = {
+            serializer: serializer.Meta.model for serializer in polymorphic_serializers}
+        model_to_serializer = {
+            serializer.Meta.model: serializer for serializer in polymorphic_serializers}
+        type_to_serializer = {
+            get_resource_type_from_serializer(serializer): serializer for
+            serializer in polymorphic_serializers}
+        setattr(new_class, '_poly_serializer_model_map', serializer_to_model)
```

Comment:
No reasons actually: first I was considering formatting the name depending on the serializer but it would have been pointless and overkill. Later, I used `setattr` for consistency...